tests/test_api_sources.py

```diff
@@ -265,44 +282,118 @@ def test_sources_cru_existing(session):
         session.create_source(SOURCE_ID_WAVEFORM, meta)
 
 
-def test_tabular_sources(session):
+def test_tabular_sources(session, tabular_source):
     columns = tabular_df.columns.tolist()
     columns.remove("timestamp")
     meta = {"extra": "information"}
-    session.create_tabular_source(SOURCE_ID_TABULAR, meta, columns)
-    src = session.get_source(SOURCE_ID_TABULAR)
-    assert src["source_id"] == SOURCE_ID_TABULAR
+    src = session.get_source(tabular_source)
+    assert src["source_id"] == tabular_source
     assert src["meta"] == meta
     assert src["properties"]["columns"] == columns
 
 
-def test_tabular_measurements_float_timestamps(session):
+def test_tabular_measurements_float_timestamps(session, tabular_source):
     with pytest.raises(HTTPError) as exc:
         tabular_dict_float = tabular_dict.copy()
         tabular_dict_float["timestamp"] = [ts + 0.1 for ts in tabular_dict["timestamp"]]
-        session.create_tabular_measurement(SOURCE_ID_TABULAR, tabular_dict_float)
+        session.create_tabular_measurement(tabular_source, tabular_dict_float)
     assert exc.value.response.status_code == 422
 
 
-def test_tabular_measurements(session):
+def test_tabular_measurements(session, tabular_source):
     columns = tabular_df.columns.tolist()
     columns.remove("timestamp")
-    session.create_tabular_measurement(SOURCE_ID_TABULAR, tabular_dict)
+    session.create_tabular_measurement(tabular_source, tabular_dict)
 
     ts0 = tabular_dict["timestamp"][0]
-    meas = session.read_single_measurement(SOURCE_ID_TABULAR, ts0)
+    meas = session.read_single_measurement(tabular_source, ts0)
     assert meas["meta"] == {}
     assert all(np.array(meas["data"]) == tabular_df.drop("timestamp", axis=1).iloc[0])
     assert meas["columns"] == columns
 
-    session.update_measurement(SOURCE_ID_TABULAR, ts0, {"new": "meta"})
-    meas = session.read_single_measurement(SOURCE_ID_TABULAR, ts0)
+    session.update_measurement(tabular_source, ts0, {"new": "meta"})
+    meas = session.read_single_measurement(tabular_source, ts0)
     assert meas["meta"] == {"new": "meta"}
 
-    session.delete_measurement(SOURCE_ID_TABULAR, ts0)
+    session.delete_measurement(tabular_source, ts0)
     with pytest.raises(HTTPError) as exc:
-        session.read_single_measurement(SOURCE_ID_TABULAR, ts0)
+        session.read_single_measurement(tabular_source, ts0)
     assert exc.value.response.status_code == 404
 
 
+def test_create_label(session, tabular_source_with_measurements):
+    timestamps = tabular_dict["timestamp"]
+    label1 = {"label": "normal", "severity": 0, "notes": ""}
+    label2 = {"label": "failure", "severity": 100, "notes": "This is really bad!"}
+
+    session.create_label(tabular_source_with_measurements, timestamps[0], **label1)
```

Comment:
Using keyword arguments here is a sign that the method signature is too complex, rather stick to the API spec (see my comment on "severity" above.